src/game-logic/reducers/harvestPlot.js

```diff
@@ -96,3 +96,46 @@ export const harvestPlot = (state, x, y) => {
     },
   }
 }
+
+/**
+ * @param {farmhand.state} state
+ * @param {number} x
+ * @param {number} y
+ * @returns {farmhand.state}
+ */
+function harvestWeed(state, x, y) {
+  const row = state.field[y]
+  const crop = row[x]
+  const item = itemsMap[crop.itemId]
+  const harvestedQuantity = getHarvestedQuantity(state)
+
+  state = removeFieldPlotAt(state, x, y)
+  state = addItemToInventory(state, item, harvestedQuantity)
+
+  return state
+}
+
+/**
+ * @param {farmhand.state} state
+ * @param {number} x
+ * @param {number} y
+ * @returns {farmhand.state}
+ */
+export const harvestPlot = (state, x, y) => {
```

Comment:
The new changes in this file make me really glad we moved each reducer to its own module!